src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -564,18 +564,37 @@ private int EndReadWrite(IAsyncResult asyncResult)
                     throw new ArgumentException();
                 }
 
-                DtrEnable = dtrEnable;
+                try
+                {
+                    DtrEnable = dtrEnable;
+                }
+                catch (IOException)
```

Comment:
Thanks for making the change, @sleeuwen . This fix is on pair with @krwq suggestion in the issue to catch the exception.
@krwq @wfurt is it possible to unit test these changes?

Reply:
This constructor is internal. It might be a good idea to document the new behavior in the APIs that consume it.

Reply:
I'm wondering if we should swallow the exception only when `dtrEnable=false`. I see no reason to fail in that case. However when somebody as asking for DTR I'm not sure if it is ok to silently ignore the request. 

As far as the testing @carlossanlop : we currently do not have permanent setup with serial hardware and this is also HW specific e.g. difficult to test in general environment.

Reply:
@carlossanlop I'm not sure I understand what you're asking with documenting the new behavior. There is only one consumer as far as I can see, and that is the `SerialPort` in it's `Open()` method.

@wfurt I'm fine with only swallowing the exception if `dtrEnable=false`, but before I do I'd like confirmation that's what needs to be done.

Reply:
@wfurt Can you confirm that we should only swallow the exception when `dtrEnable=false`?

Reply:
I agree with @wfurt swallowing an exception is only acceptable when dtrEnable=false and similarly for rts. Even better solution could be to try to see what native call returns and avoid throwing/catching completely